### PR TITLE
dsp_dsp.cpp: fix registering Interrupt handler on loading savestates

### PR DESCRIPTION
**Pull request by SachinVin**
creating a shared pointer from the serialize function proved rather difficult, 
So I just moved this to the constructor...

@Steveice10 @tywald please check if this works on your end

Closes https://github.com/citra-emu/citra/issues/7047.

- - -
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/7055)

### CODE REVIEW

**Comment by Steveice10** [comment]
Works on my end, thanks. Makes sense to me too as this is how most services register things like timing callbacks IIRC.

**Review by Steveice10 — APPROVED** [review]

File: src/core/hle/service/dsp/dsp_dsp.cpp [hunk 1]
```diff
@@ -22,7 +22,6 @@ enum class DspPipe;
 }
 
 namespace Service::DSP {
-
```
Comment (FearlessTobi) [inline src/core/hle/service/dsp/dsp_dsp.cpp:25]: seems like this was removed on accident?

**Review by FearlessTobi — APPROVED** [review]
LGTM. Does this fix https://github.com/citra-emu/citra/issues/7047 ?

**Comment by Steveice10** [comment]
> LGTM. Does this fix #7047 ?

Yes, that's what I tested for.

**Pull request merged**